tools/mo/openvino/tools/mo/convert_impl.py

```diff
@@ -881,19 +873,46 @@ def _convert(**args):
                 args['onnx_opset_version'] = None
 
                 try:
-                    ov_model = _convert(**args)
+                    ov_model = _convert(cli_parser, framework, args)
                 except Exception as e:
                     remove_tmp_onnx_model(out_dir)
                     raise e
 
                 remove_tmp_onnx_model(out_dir)
                 return ov_model
-        args = params_to_string(**args)
-        argv, non_default_params = pack_params_to_args_namespace(**args)
+
+        if len(args) > 0:
+            args_string = params_to_string(**args)
+            argv = cli_parser.parse_args(args_dict_to_list(cli_parser, **args_string))
+
+            # get list of all available params for convert_model()
+            all_params = {}
+            for key, value in mo_convert_params.items():
+                all_params.update(value)
+
+            # check that there are no unknown params provided
+            for key, value in args_string.items():
+                if key not in argv and key not in all_params.keys():
+                    raise Error("Unrecognized argument: {}".format(key))
+
+                # Non string params like input_model or extensions are ignored by parse_args()
+                # so we need to set them in argv separately
+                if value is not None and getattr(argv, key) != value:
+                    setattr(argv, key, value)
+        else:
+            argv = cli_parser.parse_args()
+
+        if framework is not None:
+            setattr(argv, 'framework', framework)
```

Comment:
I think we need to move this into separate routine like `pack_params_to_args_namespace` and test it well.